op-batcher/batcher/driver.go

```diff
@@ -345,13 +350,7 @@ func (l *BatchSubmitter) loop() {
 	for {
 		select {
 		case <-ticker.C:
-			if txpoolState.CompareAndSwap(TxpoolBlocked, TxpoolCancelPending) {
-				// txpoolState is set to Blocked only if Send() is returning
-				// ErrAlreadyReserved. In this case, the TxMgr nonce should be reset to nil,
-				// allowing us to send a cancellation transaction.
-				l.cancelBlockingTx(queue, receiptsCh, txpoolBlockedBlob)
-			}
-			if txpoolState.Load() != TxpoolGood {
+			if !l.checkTxpool(queue, receiptsCh) {
 				continue
 			}
```

Comment:
Now that you put `l.checkTxpool` into `publishStateToL1`, I'm wondering if we even need this check here still. It doesn't hurt to call `loadBlocksIntoState` if the txpool already got flagged bad, and even the `publishAndWait()` call within the reorg branch would call `publishStateToL1`, which then does the txpool check.

Reply:
I think that's right, it's not really needed, but I think it does add clarity?  I originally had publishStateToL1 just check the state and abort without issuing the cancel, leaving cancelling for the loop(), but then noticed reusing checkTxpool could save a couple lines of code without much drawback.

Reply:
It doesn't hurt, ok to leave it in.